### PR TITLE
Added the tests, estimates and made the general changes to One var Fit Model dialog

### LINKED ISSUE #5810: Enable 2nd button in Model > Fit Model > One Variable
This will start by including the t-test, binomial test and proportions from the Mosaic package. They generalise the same functions in the stats package.  
Here is the existing dialogue:
image

In this dialogue:
a) We change the second button from "Exact Solution" to just say "Test", and add a third button with the label "Estimate".
b) When the second button is active the label with the pull-down changes to **_Test:_**.  It is Estimate: with the third button.
c) Similarly, the Save Model label becomes **_Save Test_** and **_Save Estimate_**.  The field to save the name can move right slightly, so it remains in the same position for all options - and there is enough room for the Save Estimate label.
d) Similarly the default object names are **_test1_** and **_ci1_**, respectively rather than dist1.
e) Similarly the **_Display Options_** and **_Fitting Options_** buttons are hidden.

Now: with the second button, **_now labelled Test_**.  The pull-down menu (which is there and labelled Dist above) has the following tests (at least for now)
1)  **_binomial_** (calls mosaic::binom.test)
2) **proportion**  (calls mosaic:prop.test)
3) **_sign_** (probably calls DescTools::SignTest) 
4) **_t_** (calls mosaic::t.test)
5) Wilcoxon (calls wilcoxon.test)
6) **_Z_** (calls DescTools::ZTest)
### Now some tests for whether the data can be taken as a random sample:
**_A separator first_**.  Then
1) **_Bartel_** (calls DescTools::BartelsRankTest with tooltip: Bartels Rank Test of Randomness)
2) **_br_** (Calls trend::br.test, with tooltip: Buishand Range Test for Change-Point Detection
3) **_runs_** (calls DescTools::RunsTest)
4) **_Sen_** (calls:trend::sens.slope with tooltip: Sen’s slope for linear rate of change) 
4) **_serial corr_** (calls EnvStats::serialcorrelationTest)
4) **_snh_** (Calls trend::snh.test with tooltip: Standard Normal Homogeinity Test (SNHT) for Change-Point Detection)
### Now some tests for normality - all from nortest package
**_Another separator_**
1) **_ad_** (With tooltip: Anderson-Darling test for normality - nortest::ad.test)
2) **_cvm_** (with tooltip: Cramer-von Mises test for normality - nortest::cvm.test)
3) **_lillie_** (with tooltip: Lilliefors (Kolmogorov-Smirnov) test for normality - nortest::lillie.test)
4) **_Pearson_** (with tooltip: Pearson chi-square test for normality - nortest::pearson.test)
5) **_sf_** (with tooltip: Shapiro-Francia test for normality - nortest::sf.test)

With the third button - the new **_Estimate_** button.  Now change the Dist label to **_Estimate_**.  Then the pull-down now has the following options:
1) mean (Calls MeanCI from DescTools)
2) median (Calls MedianCI from DescTools)
3) normal (Calls normCI from MKinfer)
4) poisson (Calls PoissonCI from DescTools)
5) quantile (Calls quantileCI from MKinfer)
6) sd (calls sdCI from MKinfer)
7) variance (Calls VarCI from DescTools)
 
### The final part
Each of these options will have further simple controls under the single receiver.
They will be similar, so I i9nitially give the controls for one or two from each of the new sections:

1) mean from meanCI in DescTools.  Here is the command from DescTools:
MeanCI(x, sd = NULL, trim = 0, method = c("classic", "boot"),conf.level = 0.95, sides = c("two.sided", "left", "right"),na.rm = FALSE, ...)
a) Ignore sd
b) Trim That is between 0 and 0.5.  That's an up-down, with default 0 and max 0.5 in steps of 0.01.  (We already have an example in the Prepare > Data: Reshape > Column Summaries under the More tab.)
c) Method Is that a simple pull down or radio buttons, with the options classic or boot.  Default is Classic.
d) Confidence Level;  Do we already have this anywhere already?  The default is (as in the dialogue) 0.95, other values are 0.8, 0.9, 0.95 (default), 0.99, 0.999, and ideally you can type anything between 0 and 1.  This control is in almost all the options here.
e) Ignore sides - so it always uses the default, which is 2-sided.  This is to be ignored generally.
f) Ignore the na.rm, but include na.rm=TRUE in the resulting R command.

So for this example there are just 3 simple controls.
I hope you can use these "rules" for more examples too.

### PR DESCRIPTION
**Pull request by Wycklife**
Fixes #5810

@africanmathsinitiative/developers  This is still in progress.

@shadrackkibet Is working on the general code structure for this dialog before I proceed with my work.

### CODE REVIEW

**Comment by shadrackkibet** [comment]
@Wycklife I am now taking up this to tidy the code and merge it into your branch. Thanks.

**Comment by rdstern** [comment]
@Wycklife tell me when you are ready for me to examine.  And are you still just working on the first 6 tests?  If not, then you could edit the name of this pull request.

**Comment by lloyddewit** [comment]
@Wycklife Please could you also ensure that this PR addresses all of the historical PR #3892 changes, thanks

**Comment by Wycklife** [comment]
> @Wycklife Please could you also ensure that this PR addresses all of the historical PR #3892 changes, thanks

Okay, I  will do that

**Comment by shadrackkibet** [comment]
> @Wycklife tell me when you are ready for me to examine. And are you still just working on the first 6 tests? If not, then you could edit the name of this pull request.

@Wycklife I am just looking at this from a distance. Good that you have been working hard on this. It looks like there has been a lot of work going on because I see now you edited the name of this PR meaning you are adding all tests? and there has been a substantial number of commits. I think it will be appropriate for @rdstern to examine this before you continue. Do you agree?

**Comment by rdstern** [comment]
Now I reset and run the binomial.  I put the value 2 into the null hypothesis - which is impossible (it should be more than 0 and less than 1.  It remains ok.
There is no other control, so I press ok.  I now see that it has anyway ignored the value I put in and tested h0 that p= 0.5!  So your test (of binomial anyway) is not picking up the value of the null typothesis.
I also deliberately ran a silly example, using the yield variable.  As I mentioned in a previous comment to you, this is an example where you do need an extra control which says what is to be a success.  Here is the documentation for the binomial in mosaic
image

You notice there is an argument that will need a control called **_Success_**. 
In the documentation above I see that it would be good to also have another pull-down control with label **_C.I. Method:_**.

(I was ok with you spending so long, because I assumed this is what you were doing for each option.   It is disappointing to see how much clearly remains to be done.)  Please also get @Ivanluv or @shadrackkibet to check the code as well as the working of the different options.  And keep me in touch.

**Comment by Wycklife** [comment]
@rdstern I have not finished working on the dialog, I have been mostly working on the tests and estimate functions, setting up the controls was to be my last part of the work since it's a small task. I noticed all that you've mentioned about the dialog.

**Comment by rdstern** [comment]
@Wycklife and I also missed your request to examine the estimates part of the dialogue - which looks much better.  I saw the word test and went straight for the tests part.
On the estimates:
a) Delete the poisson case.  It doesn't work as I thought it did.
b) The Confidence Level control should be a combo box into which you can also type - not an up-down.  It has the values 0.98, 0.9, 0.95, 0.99 and 0.999 and you can type yourself anything between 0.5 and 0.999.
c) Delete the **Trim control** (or hide it for now.  It currently is still there also for the median - when it doesn't apply.  Simplest to omit it also for the mean.
d) The quantile option needs an extra control where you give the quantile.  Anything more than 0 and less than 1 is allowed.  It could be an up-down into which you can type which goes from 0.05 to 0.95 and default is 0.5.  
e) In the quantile option please check the methods.  I think the options are exact or asymptotic.
f) You have Omit Missing (wrongly spelled).  You could change it to Omit Missing Values and put it under the Estimate control, so it is in the same place as in the Prepare > Data Reshape > Column Summaries dialogue.  It seems checxked by default, so make that the default.  Then it doesn't seem to be connected yet.  So connect.

**Comment by Wycklife** [comment]
@rdstern I have made some improvements to this pr.  Most of the tests, estimates and controls are okay except the following which will be implemented by tomorrow, as well as some requires discussions with you possibly on Monday. 
1. implementing proportion and binomial tests using the success argument. Should we discuss this on Monday?
2. I am looking forward to finish up on the general functionality of the dialog. there are few things that needs to be set i.e test ok , reopening of the dialog and a ligning up of the controls. For now just check whether the tests and estimates are working well. Possibly everything should be set up by tomorrow except binomial and the proportion tests.

**Comment by rdstern** [comment]
@Wycklife I need you to do more checking of the code and called first.
I started with Mean and then Median under estimates and used the yield variable from the usual survey data.  Here were the commands:  (And both gave errors)
My suggestions were as follows:

mean (Calls MeanCI from DescTools)
median (Calls MedianCI from DescTools)

Instead your code calls fitdistrplus for the mean and mosaic::binomial for the median.


```
# Code generated by the dialog, One Variable Fit Model
yield <- data_book$get_columns_from_data(data_name="survey", col_names="yield")
last_model <- fitdistrplus::fitdist(data=as.vector(x=stats::na.exclude(object=yield)), method="mle")
data_book$add_model(model_name="last_model", model=last_model, data_name="survey")
data_book$get_models(data_name="survey", model_name="last_model")

rm(list=c("last_model", "yield"))

# Code generated by the dialog, One Variable Fit Model
yield <- data_book$get_columns_from_data(data_name="survey", col_names="yield")
last_model <- mosaic::binom.test(x=yield, p="0.5", ci.method="Clopper-Pearson")
data_book$add_model(model_name="last_model", model=last_model, data_name="survey")
data_book$get_models(data_name="survey", model_name="last_model")

rm(list=c("last_model", "yield"))
```

**Comment by Wycklife** [comment]
@rdstern  This has been completed, except for binomial and the proportion tests. Kindly check whether the tests and estimates are working as expected.

**Comment by Wycklife** [comment]
@rdstern  I didn't find a way of adding a separator inside acombobox.

**Comment by rdstern** [comment]
I am checking again @wyclife and (at last) this seems promising.

On the control for the confidence interval you still have to make the changes.

a) See my suggestions 6 days ago on the control for the confidence level.  You have changed it, so it is now between 0 and 1, but see my suggestion above.  It should not be an up-down. And I tried 0 and 1 currently and it gives an error.
I see you have correctly changed it for the estimates.
b) And for that control you have correctly allowed me to type in.  But whatever I type gives an error.  Did you test this?
c) And I can type values more than 1.  But oh it (correctly traps this and gives me a message.  Unfortunately the message is wrong.  It says >- 0 and < 1.  It should  be > 0 and < 1. 
d) z test works, but the st dev for the population can be 0 or negative, which gives an error.  Restrict it to being >0.
e) Look again at my message about positioning the Omit Missing Values checkbox of 6 days ago and also look where the same control is, in the Prepare > Data Reshape > Column Summaries dialogue.  It could usefully go in the same place as a I suggested before.
f) In quantiles add a control for method - see the documentation - with the options exact, asymptotic and boot.  Only show the control boot.ci control when the method control is set to boot.  In the bootci options don't have capital letters for the options
g) Look again at the options for both tests and estimates in the original issue.  Is there a good reason that you have each option starting with a capital letter?  Otherwise change them back to what was suggested.

**Comment by rdstern** [comment]
@wyclife.  Good last comment.  Says exactly what the change in the dialogue will be.

**Comment by Wycklife** [comment]
@rdstern Could you check the position of the omit Missing Values checkbox. Below is the screen shot

image

**Comment by Wycklife** [comment]
@rdstern I have fixed the above changes.

**Comment by rdstern** [comment]
ok

**Comment by lloyddewit** [comment]
@Wycklife I'm not sure if you have fully tested this, and whether it's ready for test/review by others. Please could you clarify the status? thanks

**Comment by Wycklife** [comment]
> @Wycklife I'm not sure if you have fully tested this, and whether it's ready for test/review by others. Please could you clarify the status? thanks

Sorry for late reply, this is what am currently working on. Once it's done I will let you know.

**Comment by rdstern** [comment]
@Wycklife I saw you added the separator - if so, that's great.

**Comment by Wycklife** [comment]
@Ivanluv @shadrackkibet and @rdstern Please could you review this?

**Comment by Ivanluv** [comment]
The MKinfer and Trend packages seem not to available at first when testing and so they have to be installed first.Will this be a problem?

File: instat/dlgOneVarFitModel.vb [hunk 1]
```diff
@@ -42,287 +50,513 @@ Public Class dlgOneVarFitModel
...
-        ucrReceiverVariable.SetParameter(New RParameter("object"))
+        ucrReceiverVariable.SetParameter(New RParameter("object", 0))
+        ucrReceiverVariable.Selector = ucrSelectorOneVarFitMod
         ucrReceiverVariable.SetParameterIsRFunction()
+        ucrReceiverVariable.SetMeAsReceiver()
 
         ucrChkConvertVariate.SetText("Convert to Numeric")
         ucrChkConvertVariate.AddParameterValueFunctionNamesCondition(True, "data", "as.numeric", True)
+        ucrChkConvertVariate.AddParameterValueFunctionNamesCondition(False, "data", "as.numeric", False)
         ucrChkConvertVariate.AddParameterValueFunctionNamesCondition(False, "data", frmMain.clsRLink.strInstatDataObject & "$get_columns_from_data", True)
```
Comment (Ivanluv) [inline instat/dlgOneVarFitModel.vb:84]: I don't think this is being used

File: instat/dlgOneVarFitModel.vb [hunk 2]
```diff
@@ -42,287 +50,513 @@ Public Class dlgOneVarFitModel
...
         clsRplotPPComp = New RFunction
         clsRplotCdfcomp = New RFunction
         clsRplotQqComp = New RFunction
         clsRplotDenscomp = New RFunction
         clsRLogLikFunction = New RFunction
-        clsFamilyFunction = New RFunction
-        clsROneVarFitModel = New RFunction
+
         ucrDistribution = New ucrDistributions
+        ucrChkOmitMissing.Checked = True
```
Comment (Ivanluv) [inline instat/dlgOneVarFitModel.vb:271]: This is not needed

File: instat/dlgOneVarFitModel.vb [hunk 3]
```diff
@@ -543,191 +606,316 @@ Public Class dlgOneVarFitModel
...
     Private Sub ucrPnlGeneralExactCase_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrPnlGeneralExactCase.ControlValueChanged
-        EnableOptions()
-        BinomialConditions()
-        SetDistributions()
-        DataTypeAccepted()
+        If rdoGeneralCase.Checked Then
+            ucrDistributionChoice.SetAllDistributions()
+        Else
+            ucrDistributionChoice.SetExactDistributions()
+        End If
```
Comment (Ivanluv) [inline instat/dlgOneVarFitModel.vb:642]: the else statement might not be needed.@shadrack could help us with this

File: instat/dlgOneVarFitModel.vb [hunk 4]
```diff
@@ -16,22 +16,30 @@
...
-    Public clsFunctionOperator, clsFactorOperator As New ROperator
+    Private clsROneVarFitModel, clsRLogLikFunction, clsRfitdist, clsRConvertVector, clsRConvertInteger, clsRConvertNumeric, clsNaExclude As New RFunction
+    Private clsRplotFunction, clsRplotPPComp, clsRplotCdfcomp, clsRplotQqComp, clsRplotDenscomp As New RFunction
+    Private clsBionomialFunction, clsProportionFunction, clsSignTestFunction, clsTtestFunction, clsWilcoxonFunction, clsZTestFunction, clsBartelFunction, clsBrFunction, clsRunsFunction, clsSenFunction, clsSerialCorrFunction, clsSnhFunction, clsAdFunction, clsCvmFunction, clsLillieFunction, clsPearsonFunction, clsSfFunction As New RFunction
+
+    Private Sub ToolTip2_Popup(sender As Object, e As PopupEventArgs)
+
+    End Sub
+
+    Private Sub ToolTip1_Popup(sender As Object, e As PopupEventArgs) Handles tttests.Popup
```
Comment (Patowhiz) [inline instat/dlgOneVarFitModel.vb:27]: Empty sub Probably no longer needed?

File: instat/dlgOneVarFitModel.vb [hunk 5]
```diff
@@ -16,22 +16,30 @@
...
-    Public clsRplotFunction, clsRplotPPComp, clsRplotCdfcomp, clsRplotQqComp, clsRplotDenscomp As RFunction
-    Public clsFunctionOperator, clsFactorOperator As New ROperator
+    Private clsROneVarFitModel, clsRLogLikFunction, clsRfitdist, clsRConvertVector, clsRConvertInteger, clsRConvertNumeric, clsNaExclude As New RFunction
+    Private clsRplotFunction, clsRplotPPComp, clsRplotCdfcomp, clsRplotQqComp, clsRplotDenscomp As New RFunction
+    Private clsBionomialFunction, clsProportionFunction, clsSignTestFunction, clsTtestFunction, clsWilcoxonFunction, clsZTestFunction, clsBartelFunction, clsBrFunction, clsRunsFunction, clsSenFunction, clsSerialCorrFunction, clsSnhFunction, clsAdFunction, clsCvmFunction, clsLillieFunction, clsPearsonFunction, clsSfFunction As New RFunction
+
+    Private Sub ToolTip2_Popup(sender As Object, e As PopupEventArgs)
```
Comment (Patowhiz) [inline instat/dlgOneVarFitModel.vb:23]: Empty sub, Probably no longer needed?

**Review by Patowhiz — COMMENTED** [review]
Would recommend comments addition to explain certain parts of the code that are not obvious

**Comment by Ivanluv** [comment]
@Patowhiz  am taking over this

**Comment by lloyddewit** [comment]
This PR was replaced by PR #6004 and PR #6150

**Pull request closed without merging**